src/config/BaseConfig.ts

```diff
@@ -0,0 +1,26 @@
+import ICloudConfig from "../interfaces/Config/ICloudConfig";
+import {INVALID_TYPE_MESSAGE} from "../constants";
+
+class Config {
+  /**
+   * Type guard for user config input
+   * @param {any} object User provided input of unknown shape
+   * @param {string[]} validKeys Array of valid configuration keys
+   * @return {boolean} Whether an object is a valid configuration object
+   */
+  isValidConfig(object:unknown, validKeys:string[]): object is ICloudConfig {
+    if (typeof object !== 'object' || object instanceof Array) {
+      return false;
+    }
+
+    Object.keys(object).forEach((key) => {
```

Comment:
Don't understand how this validation goes together with the `[other:string]: unknown;` in all interfaces